metals/src/main/scala/scala/meta/internal/metals/ammonite/Ammonite.scala

```diff
@@ -239,35 +240,41 @@ final class Ammonite(
           if (docs.isEmpty) "No Ammonite script is opened"
           else "No open document is not an Ammonite script"
         scribe.error(msg)
-        Future.failed(new Exception(msg))
+        Future.failed(new AmmoniteMetalsException(msg))
```

Comment:
Also, previously we created this `msg` here, bu this would never be shown to the user. Now if this ever happens, it is caught in the `recoverWith` down below which was added.